src/agoo.c

```diff
@@ -233,20 +233,27 @@ agoo_load_graphql(agooErr err, const char *path, const char *filename) {
 	return agoo_err_no(err, "Failed to open %s.", filename);
     }
     if (0 != fseek(f, 0, SEEK_END)) {
+      fclose(f);
 	return agoo_err_no(err, "Failed to seek %s.", filename);
     }
     if (0 > (size = ftell(f))) {
+      fclose(f);
 	return agoo_err_no(err, "Failed to ftell %s.", filename);
     }
     rewind(f);
     if (0 < size) {
 	if (NULL == (sdl = AGOO_MALLOC(size))) {
+    fclose(f);
+    free(sdl);
```

Comment:
No need for `free()` here as the allocation failed.

Reply:
Hi @ohler55, thank you for the time spent!

Fix in 06f1201c21faab47c5f9d1e13611a778be666131